src/app/modules/group/http-services/granted-permissions.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import * as D from 'io-ts/Decoder';
+import { decodeSnakeCase } from '../../../shared/operators/decode';
+import { dateDecoder } from '../../../shared/helpers/decoders';
+import { appConfig } from '../../../shared/helpers/config';
+import { HttpClient, HttpParams } from '@angular/common/http';
+
+const groupPermissionsDecoder = D.struct({
+  canEdit: D.literal('none', 'children', 'all', 'all_with_grant'),
+  canEnterFrom: dateDecoder,
+  canEnterUntil: dateDecoder,
+  canGrantView: D.literal('none', 'enter', 'content', 'content_with_descendants', 'solution', 'solution_with_grant'),
+  canMakeSessionOfficial: D.boolean,
+  canView: D.literal('none', 'info', 'content', 'content_with_descendants', 'solution'),
+  canWatch: D.literal('none', 'result', 'answer', 'answer_with_grant'),
+  isOwner: D.boolean,
+});
```

Comment:
This decoder is very similar to the one in group permissions http service, I think the decoder could be shared between group http service like it's done in item http service (for permissions as well)

Reply:
Fixed